File: others/yksdyWeb/libs/yahoo web service-2.11/yws-2.11/Perl/simpleSearch.pl
#!/usr/bin/env perl

# Author: Premshree Pillai
# http://premshree.livejournal.com

use LWP::Simple;

my $url = "http://search.yahooapis.com/WebSearchService/V1/webSearch?appid=YahooDemo&query=madonna&results=2";

# make the request
my $results = get $url || die "Web services request failed: $!";

my %entities = ( "<"=>"&lt;", ">"=>"&gt;", "&"=>"&amp;", "'"=>"&#039;", '"'=>"&quot;" );
while (my ($k, $v) = each(%entities)) {
	$results =~ s/$k/$v/g;
}
print $results;

File: others/yksdyWeb/libs/yahoo web service-2.11/yws-2.11/Ruby/simpleSearch.rb
#!/usr/bin/env ruby

# Author: Premshree Pillai
# http://premshree.livejournal.com

require "net/http"

request = "http://search.yahooapis.com/WebSearchService/V1/webSearch?appid=YahooDemo&query=madonna&results=2"

# make the request
begin
	results = Net::HTTP.get_response(URI.parse(request)).body
rescue
	raise "Web services request failed"
end
{"<"=>"&lt;", ">"=>"&gt;","&"=>"&amp;",'"'=>"&quot;", "'"=>"&#039;"}.each { |k,v|
	results.gsub!(k,v)
}

print results; 

